### PR TITLE
test without regex

### PR DESCRIPTION
**Pull request by ekoeryanto**

### CODE REVIEW

**Comment by mcollina** [comment]
Why this change? How did you notice?

File: jwt.js [hunk 1]
```diff
@@ -192,7 +192,7 @@ function fastifyJwt (fastify, options, next) {
         let scheme = parts[0]
         token = parts[1]
 
-        if (!/^Bearer$/i.test(scheme)) {
+        if (scheme !== 'Bearer') {
```
Comment (mcollina) [inline jwt.js]: Note that the regexp matches `'bearer'` as well. Is this an expected change? I think http headers are meant to be case insensitive.

cc @jsumners

Comment (ekoeryanto) [reply jwt.js]: uh, i miss the `i` flag.

**Comment by ekoeryanto** [comment]
just guessed that regex is slower

File: jwt.js (hunk 1 above)
Comment (jsumners) [reply jwt.js]: Typically it’s only the header names that are case insensitive. Whether or not the values are insensitive is up to the server implementation.

**Comment by cemremengu** [comment]
Maybe we can add a unit test for case insensitivity?

**Comment by jsumners** [comment]
According to spec, the value should be `Bearer foo`.

https://tools.ietf.org/html/rfc6750#section-2.1
> When sending the access token in the "Authorization" request header
   field defined by HTTP/1.1 [RFC2617], the client uses the "Bearer"
   authentication scheme to transmit the access token.
>
>   For example:
>
>     GET /resource HTTP/1.1
>     Host: server.example.com
>     Authorization: Bearer mF_9.B5f-4.1JqM
>
>   The syntax of the "Authorization" header field for this scheme
   follows the usage of the Basic scheme defined in Section 2 of
   [RFC2617].  Note that, as with Basic, it does not conform to the
   generic syntax defined in Section 1.2 of [RFC2617] but is compatible
   with the general authentication framework being developed for
   HTTP 1.1 [HTTP-AUTH], although it does not follow the preferred
   practice outlined therein in order to reflect existing deployments.
   The syntax for Bearer credentials is as follows:
>
>     b64token    = 1*( ALPHA / DIGIT /
>                       "-" / "." / "_" / "~" / "+" / "/" ) *"="
>     credentials = "Bearer" 1*SP b64token
>
>   Clients SHOULD make authenticated requests with a bearer token using
   the "Authorization" request header field with the "Bearer" HTTP
   authorization scheme.  Resource servers MUST support this method.

**Comment by ekoeryanto** [comment]
@cemremengu except for test coverage reason, i think it doesn't need to

**Comment by mcollina** [comment]
Then it's my faul, let's move it back to `Bearer` only.

**Comment by cemremengu** [comment]
Should we close this? Or just change the check to be `=== 'Bearer` ?

**Comment by mcollina** [comment]
The second one @cemremengu.

**Pull request closed without merging**